src/core/api-session.utils.ts

```diff
@@ -0,0 +1,65 @@
+import { AppEvents, DataSourceInstanceSettings, EventBus } from "@grafana/data";
+import { BackendSrv, getAppEvents } from "@grafana/runtime";
+import { post } from "./utils";
+
+export interface ApiSession {
+    endpoint: string;
+    session: {
+        expiry: string,
+        secret: string
+    };
+}
+
+export class ApiSessionUtils {
+    private readonly appEvents: EventBus;
+    private readonly cacheExpiryBufferTimeInMilliseconds = 5 * 60 * 1000; // 5 minutes buffer
+    private sessionCache?: ApiSession;
```

Comment:
```suggestion
    private static _sessionCache?: Promise<ApiSession>;
```

Refer to existing utils classes and update the methods.

Reply:
Refer to this [comment ](https://github.com/ni/systemlink-grafana-plugins/pull/226#discussion_r2130632778)for more info.